src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -0,0 +1,284 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem; // Roslyn
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+
+/// <summary>
+/// Entry point for the hosting of Roslyn language services by the .NET project system.
+/// One instance of this host exists per unconfigured project.
+/// </summary>
+/// <remarks>
+/// The core responsibilities of this host are:
+///
+/// <list type="bullet">
+///   <item>
+///     To create, populate and destroy instances of <see cref="IWorkspaceProjectContext"/> during project load,
+///     evaluation/build, close and changes to the set of project configurations.
+///   </item>
+///   <item>
+///     To manage operation progress registrations so that IDE features wait for the language service to be initialized,
+///     preventing things like error squigglies appearing during project load.
+///   </item>
+///   <item>
+///     To interrogate the active workspace as part of various IDE features, including acquiring a read or write lock.
+///   </item>
+/// </list>
+/// </remarks>
+[Export(typeof(IWorkspaceWriter))]
+[Export(ExportContractNames.Scopes.UnconfiguredProject, typeof(IProjectDynamicLoadComponent))]
+[AppliesTo(ProjectCapability.DotNetLanguageService)]
+internal sealed class LanguageServiceHost : OnceInitializedOnceDisposedUnderLockAsync, IProjectDynamicLoadComponent, IWorkspaceWriter
+{
+    // TODO don't activate in if _vsShellServices.Value.IsInCommandLineMode
```

Comment:
What is this in reference to?

Reply:
https://github.com/dotnet/project-system/issues/3832

I'll include that in the code directly too.